internal/consensus/reactor.go

```diff
@@ -341,63 +343,32 @@ func (r *Reactor) GetPeerState(peerID types.NodeID) (*PeerState, bool) {
 	return ps, ok
 }
 
-func (r *Reactor) broadcastNewRoundStepMessage(rs *cstypes.RoundState) {
-	select {
-	case r.stateCh.Out <- p2p.Envelope{
-		Broadcast: true,
-		Message:   makeRoundStepMessage(rs),
-	}:
-	case <-r.closeCh:
-		return
-	}
-}
-
-func (r *Reactor) broadcastNewValidBlockMessage(rs *cstypes.RoundState) {
+func (r *Reactor) makeNewValidBlockMessage(rs *cstypes.RoundState) proto.Message {
 	psHeader := rs.ProposalBlockParts.Header()
-	select {
-	case <-r.closeCh:
-		return
-	case r.stateCh.Out <- p2p.Envelope{
-		Broadcast: true,
-		Message: &tmcons.NewValidBlock{
-			Height:             rs.Height,
-			Round:              rs.Round,
-			BlockPartSetHeader: psHeader.ToProto(),
-			BlockParts:         rs.ProposalBlockParts.BitArray().ToProto(),
-			IsCommit:           rs.Step == cstypes.RoundStepApplyCommit,
-		},
-	}:
+	return &tmcons.NewValidBlock{
+		Height:             rs.Height,
+		Round:              rs.Round,
+		BlockPartSetHeader: psHeader.ToProto(),
+		BlockParts:         rs.ProposalBlockParts.BitArray().ToProto(),
+		IsCommit:           rs.Step == cstypes.RoundStepApplyCommit,
 	}
 }
 
-func (r *Reactor) broadcastHasVoteMessage(vote *types.Vote) {
-	select {
-	case <-r.closeCh:
-		return
-	case r.stateCh.Out <- p2p.Envelope{
-		Broadcast: true,
-		Message: &tmcons.HasVote{
-			Height: vote.Height,
-			Round:  vote.Round,
-			Type:   vote.Type,
-			Index:  vote.ValidatorIndex,
-		},
-	}:
+func (r *Reactor) makeHasVoteMessage(vote *types.Vote) proto.Message {
```

Comment:
since this function uses the state `types.Vote` only and doesn't touch a reactor, then want do you think to move this method into `types` package as an exportable function, put it next to Vote struct or event make a method `ToHasVoteProto()` ?
it is applicable for other factory methods too